core/src/main/java/jenkins/model/Nodes.java

```diff
@@ -0,0 +1,254 @@
+package jenkins.model;
+
+import hudson.BulkChange;
+import hudson.Util;
+import hudson.XmlFile;
+import hudson.model.Computer;
+import hudson.model.ItemGroupMixIn;
+import hudson.model.Node;
+import hudson.model.Queue;
+import hudson.model.Saveable;
+import hudson.model.listeners.SaveableListener;
+import hudson.slaves.EphemeralNode;
+import hudson.slaves.OfflineCause;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+import java.io.File;
+import java.io.FileFilter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.ConcurrentSkipListMap;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Manages all the nodes for Jenkins.
+ *
+ * @since 1.FIXME
+ */
+public class Nodes implements Saveable {
```

Comment:
Seems like this should be `@Restricted(NoExternalUse.class)` unless I am missing something?